pkg/autodiscovery/integration/config.go

```diff
@@ -37,27 +37,78 @@ const (
-	NodeName                string                 `json:"node_name"`                 // node name in case of an endpoint check backed by a pod (include in digest: true)
-	CreationTime            CreationTime           `json:"-"`                         // creation time of service (include in digest: false)
-	Source                  string                 `json:"source"`                    // the source of the configuration (include in digest: false)
-	IgnoreAutodiscoveryTags bool                   `json:"ignore_autodiscovery_tags"` // used to ignore tags coming from autodiscovery (include in digest: true)
-	MetricsExcluded         bool                   `json:"metrics_excluded"`          // whether metrics collection is disabled (set by container listeners only) (include in digest: false)
-	LogsExcluded            bool                   `json:"logs_excluded"`             // whether logs collection is disabled (set by container listeners only) (include in digest: false)
+	// When a new field is added to this struct, please evaluate whether it
+	// should be computed in the config Digest and update the field's
+	// documentation and the Digest method accordingly
+
+	// Name of the integration
+	Name string `json:"check_name"` // (include in digest: true)
+
+	// Instances is the list of instances in YAML or JSON.
+	Instances []Data `json:"instances"` // (include in digest: true)
+
+	// InitConfig is the init_config in YAML or JSON
+	InitConfig Data `json:"init_config"` // (include in digest: true)
+
+	// MetricConfig is the metric config in YAML or JSON (jmx check only)
+	MetricConfig Data `json:"metric_config"` // (include in digest: false)
+
+	// LogsConfig is the logs config in YAML or JSON (logs-agent only)
+	LogsConfig Data `json:"logs"` // (include in digest: true)
+
+	// ADIdentifiers is the list of AutoDiscovery identifiers for this
+	// integration.  If either ADIdentifiers or AdvancedADIdentifiers are
+	// present, then this config is a template and will be resolved when a
+	// matching service is discovered. Otherwise, the config will be scheduled
+	// immediately. (optional)
+	ADIdentifiers []string `json:"ad_identifiers"` // (include in digest: true)
+
+	// AdvancedADIdentifiers is the list of advanced AutoDiscovery identifiers;
+	// see ADIdentifiers.  (optional)
+	AdvancedADIdentifiers []AdvancedADIdentifier `json:"advanced_ad_identifiers"` // (include in digest: false)
+
+	// Provider is the name of the config provider that issued the config.  If
+	// this is "", then the config is a service config, representing a serivce
+	// discovered by a listener.
+	Provider string `json:"provider"` // (include in digest: false)
+
+	// Entity is the entity ID of the service, set only for service configs.
+	Entity string `json:"-"` // (include in digest: true)
+
+	// TaggerEntity is the tagger entity ID
+	// XXX how is this different from Entity?
+	TaggerEntity string `json:"-"` // (include in digest: false)
```

Comment:
Even this part? :)

Reply:
IIRC we were using the same entity format for AD and Tagger (like 2 and half years ago) the format was `<container runtime>://<id>` which is still the same format for `EntityID`. And then we started supporting APM tagging; tracers parse their container cgroups and send their container ID to the trace agent, the latter uses that container ID to query the tagger and enrich traces with container tags. The tracers don't have access to container runtime sockets and they don't know their container runtime, so they just send the <id> part, to make this work generically for any container runtime, we introduced `TaggerEntity` with the format `container_id://<id>` to simplify APM container tagging, that's why every tagger client should respect this format now.

Reply:
OK, I'll see if I can add some documentation about that.  To verify I have it right, there are currently two entity IDs formats:

 * WLM Entity:
   * `"container://<sha>"` for containers
   * `"garden_container://<sha>"` for garden containers
   * `"kubernetes_pod_uid://<uid>"` for kubernetes pods
   * `"ecs_task://<task-id>"` for ECS tasks
 * Tagger Entity:
   * `"container_id://<sha>"` for containers
   * `"container_id://<sha>"` for garden containers
   * `"kubernetes_pod://<uid>"` for kubernetes pods
   * `"ecs_task://<task-id>"` for ECS tasks
  
That seems massively unfortunate -- so close to matching, yet not! -- but what can you do?  If it's correct, I'll add it to the docs.

Reply:
I got that slightly wrong, I think -- WLM is careful to never use URL-like syntax, but AD does.  It seems that AD basically uses "Entity" as a name for a service, and that might be a good rename.  I'll update this PR to match.